submissions/xsorter/port-sniffer/program.js

```diff
@@ -0,0 +1,95 @@
+const net = require('net');
+const dns = require('dns');
+const args = require('minimist')(process.argv.slice(2));
+const range = findPortsRange();
+
+const ports = {
+  firstPort: +range[0],
+  lastPort: +range[1] ? +range[1] : +range[0],
+  openedPorts: []
+};
+
+const openedPortCheck = (host, port, checkCallback) => {
+  const socket = net.createConnection(port, host);
+  const time = socket.setTimeout(300, () => {
+    socket.destroy();
+    checkCallback(false);
+  });
+
+  socket.on('connect', () => {
+    clearTimeout(time);
+    socket.destroy();
+    process.stdout.write('.');
+    ports.openedPorts.push(port);
+    checkCallback(true);
+  });
+  socket.on('error', function () {
+    clearTimeout(time);
+    checkCallback(false);
+  });
+};
+
+const showResult = host => {
+  if (args.help) {
+    process.stdout.write(messages().help);
+    process.exit(1);
+  }
+  if (!args.host) {
```

Comment:
If no hosts -> let's ask again about hosts?

Reply:
Removed this message